eclipse-collections-code-generator/src/main/resources/impl/map/mutable/primitiveObjectHashMap.stg

```diff
@@ -1071,6 +1072,16 @@ public class <name>ObjectHashMap\<V> implements Mutable<name>ObjectMap\<V>, Exte
         return null;
     }
 
+    public Optional\<V> detectOptional(Predicate\<? super V> predicate)
+    {
+        return Optional.ofNullable(this.detect(predicate));
```

Comment:
I'm going to pull this as is, it's strictly an improvement. Let's continue to think about null handling and the how methods like detectIfNone should be implemented in the future. We may also considering adding a note about null handing to the Javadoc of these methods, explaining that they return Optional.empty() rather than throwing.